src/main/java/com/avispl/symphony/dal/avdevices/power/surgex/defenderseries/common/Util.java

```diff
@@ -0,0 +1,394 @@
+	 * @return the list of outlets, or an empty list if unavailable
+	 */
+	public static List<Outlet> getOutlets(CurrentStatus currentStatus) {
+		return Optional.ofNullable(currentStatus)
+				.map(CurrentStatus::getDevices).filter(devices -> !devices.isEmpty())
+				.map(devices -> devices.get(0).getOutlets()).filter(outlets -> !outlets.isEmpty())
+				.orElse(new ArrayList<>());
+	}
+
+	/**
+	 * Retrieves the list of outlet groups from the {@link CurrentStatus}.
+	 *
+	 * @param currentStatus the current status
+	 * @return the list of outlet groups, or an empty list if unavailable
+	 */
+	public static List<Group> getOutletGroups(CurrentStatus currentStatus) {
+		return Optional.ofNullable(currentStatus)
+				.map(CurrentStatus::getDevices).filter(devices -> !devices.isEmpty())
+				.map(devices -> devices.get(0).getGroups()).filter(group -> !group.isEmpty())
+				.orElse(new ArrayList<>());
+	}
+
+	/**
+	 * Returns the measurements of the first {@link Device}, or a new {@link DeviceMeasurements} if unavailable.
+	 *
+	 * @param devices the list of devices
+	 * @return the measurements of the first device, or default measurements
+	 */
+	private static DeviceMeasurements getMeasurements(List<Device> devices) {
+		return Optional.ofNullable(devices).filter(device -> !device.isEmpty())
+				.map(device -> device.get(0).getMeasurements())
+				.orElse(new DeviceMeasurements());
+	}
+
+	/**
+	 * Returns the first {@link EthInterface} from the list, or a new one if unavailable.
+	 *
+	 * @param ethInterfaces the list of Ethernet interfaces
+	 * @return the first Ethernet interface, or a default one
+	 */
+	private static EthInterface getFirstEthInterface(List<EthInterface> ethInterfaces) {
+		return Optional.ofNullable(ethInterfaces).filter(ethInterface -> !ethInterface.isEmpty())
+				.map(ethInterface -> ethInterface.get(0))
+				.orElse(new EthInterface());
+	}
+
+	/**
+	 * Returns the control endpoint based on the given action and current state.
+	 * <p>
+	 * Supports {@link Constant#TOGGLE} and {@link Constant#REBOOT} actions.
+	 *
+	 * @param action the control action (e.g., toggle or reboot)
+	 * @param state the current state (1 for ON, others for OFF)
+	 * @return the corresponding endpoint, or {@code null} if the action is unsupported
+	 */
+	public static String getControlEndpoint(String action, int state) {
+		if (action.equals(Constant.TOGGLE)) {
+			return state == 1 ? EndpointConstant.POWER_OFF : EndpointConstant.POWER_ON;
+		}
+		if (action.equals(Constant.REBOOT)) {
+			return EndpointConstant.REBOOT;
+		}
+		return null;
+	}
+
+	/**
+	 * Checks if the given state is rebooting {@link Outlet} or {@link Group}.
+	 *
+	 * @param state the state of the component
+	 * @return {@code true} if rebooting; otherwise {@code false}
+	 */
+	public static boolean isRebootingComponent(Integer state) {
+		return state != null && state == 2;
+	}
+
+	/**
+	 * Maps a string value to itself if not null or empty; otherwise returns null.
```

Comment:
incomplete javadoc here and for a few methods below (missing params descriptions)